addons/binding/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/handler/WindcentraleHandler.java

```diff
@@ -0,0 +1,157 @@
+
+        String urlString = BASE_URL + millId + "/live?ignoreLoadingBar=true";
+        try {
+            millUrl = new URL(urlString);
+        } catch (MalformedURLException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Constructed url '" + urlString + "' is not valid: " + e.getLocalizedMessage());
+            return;
+        }
+
+        pollingJob = scheduler.scheduleWithFixedDelay(this::updateData, 0, pollingPeriod, TimeUnit.SECONDS);
+        logger.debug("Polling job scheduled to run every {} sec. for '{}'", pollingPeriod, getThing().getUID());
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Windcentrale handler '{}'", getThing().getUID());
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    private synchronized void updateData() {
+        logger.debug("Update windmill data '{}'", getThing().getUID());
+
+        try {
+            String getMillData = getMillData();
+            JsonParser parser = new JsonParser();
+            JsonObject millData = (JsonObject) parser.parse(getMillData);
+
+            logger.trace("Retrieved updated mill data: {}", millData);
+
+            updateState(CHANNEL_WIND_SPEED, new DecimalType(millData.get(CHANNEL_WIND_SPEED).getAsString()));
+            updateState(CHANNEL_WIND_DIRECTION, new StringType(millData.get(CHANNEL_WIND_DIRECTION).getAsString()));
+            updateState(CHANNEL_POWER_TOTAL, new DecimalType(millData.get(CHANNEL_POWER_TOTAL).getAsBigDecimal()));
+            updateState(CHANNEL_POWER_PER_WD,
+                    new DecimalType(millData.get(CHANNEL_POWER_PER_WD).getAsBigDecimal().multiply(wd)));
+            updateState(CHANNEL_POWER_RELATIVE,
+                    new DecimalType(millData.get(CHANNEL_POWER_RELATIVE).getAsBigDecimal()));
+            updateState(CHANNEL_ENERGY, new DecimalType(millData.get(CHANNEL_ENERGY).getAsBigDecimal()));
+            updateState(CHANNEL_ENERGY_FC, new DecimalType(millData.get(CHANNEL_ENERGY_FC).getAsBigDecimal()));
+            updateState(CHANNEL_RUNTIME, new DecimalType(millData.get(CHANNEL_RUNTIME).getAsBigDecimal()));
+            updateState(CHANNEL_RUNTIME_PER, new DecimalType(millData.get(CHANNEL_RUNTIME_PER).getAsBigDecimal()));
+            updateState(CHANNEL_LAST_UPDATE, new DateTimeType(millData.get(CHANNEL_LAST_UPDATE).getAsString()));
+
+            if (!getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+
+        } catch (Exception e) {
+            logger.debug("Failed to update windmill data", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private String getMillData() throws IOException {
+        try (InputStream connection = millUrl.openStream()) {
```

Comment:
URLConnection is not really advised as it is sometimes a bit [difficult to manage](https://stackoverflow.com/questions/23136475/do-we-need-to-consume-httpurlconnections-error-stream-when-ioexception-thrown) regarding error handling and proxy support. I'd suggest to go for `HttpUtil.executeUrl()` instead, which ESH provides and encapsulates all code using the Jetty HTTP client.